### ISSUE TITLE
"Permission denied" - authorization issue on AWS / Cognito

### ISSUE THREAD

**Issue by rameshkemp** [kubeflow/kubeflow#5158]
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
I am trying to get authorization to work on AWS (with Cognito configured with Federated Idp using SAML ). I followed the configuration as in https://www.kubeflow.org/docs/aws/authentication-oidc/ using kfctl_aws_cognito.v1.0.2.yaml (and v1.0-branch). My spec looks like this:

```
spec:
      addons:
      - fluentd-cloud-watch
      auth:
        cognito:
            cognitoUserPoolArn: arn:aws:cognito-idp: us-east-1:xxxxx:userpool/ us-east-1_xxxxxx
            cognitoAppClientId: xxxxxbxxxxxx
            cognitoUserPoolDomain: amazon-cognito-domain
            certArn: arn:aws:acm: us-east-1:xxxxx:certificate/xxxxxxxxxxxxx-xxxx
      enablePodIamPolicy: true
      region: us-east-1
```

**What did you expect to happen:**
The authentication part seems to be working correctly, but I get this message when calling the ALB hostname:
PERMISSION_DENIED:authzadaptor-handler.handler.istio-system:

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]
AWS Cognito has been configured external federated identity provider which uses SAML. In my case the Idp is Azure AD.

**Environment:**
- Kubeflow version: v1.0.2
- kfctl version: (use `kfctl version`):  v1.0.2-0-ga476281
- Kubernetes platform: (e.g. `minikube`) AWS EKS
- Kubernetes version: (use `kubectl version`): 
Client Version: version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.6-beta.0", GitCommit:"e7f962ba86f4ce7033828210ca3556393c377bcc", GitTreeState:"clean", BuildDate:"2020-01-15T08:26:26Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.11-eks-14f01f", GitCommit:"14f01fe8f04411d5e187b220034ca2117d79f7de", GitTreeState:"clean", BuildDate:"2020-05-23T21:32:47Z", GoVersion:"go1.12.17", Compiler:"gc", Platform:"linux/amd64"}
The complete Kubeflow Manifest file:  [kf.txt](https://github.com/kubeflow/kubeflow/files/4960497/kf.txt)

**Comment by PatrickXYS** [comment]
> The authentication part seems to be working correctly, but I get this message when calling the ALB hostname:
PERMISSION_DENIED:authzadaptor-handler.handler.istio-system:

Can you provide more logs? Such as alb-ingress-controller pod, authzadaptor pod, etc.

**Comment by rameshkemp** [comment]
I got the root cause of the issue. In AWS Cognito if we have configured external federated identity provider via SAML. Then on successful authentication of the user, the user profile created in AWS cognito has the email verified attribute as false. Due to this the authzadaptor service is denying.

**Logs from pod/authzadaptor:**
> time="2020-07-27T05:50:57Z" level=warning msg="skip unknown headers: unknown"
> time="2020-07-27T05:50:57Z" level=warning msg="skip unknown headers: unknown"
> time="2020-07-27T05:50:59Z" level=error msg="Email is not verified false"
> time="2020-07-27T05:50:59Z" level=error msg="Email is not verified false"
> time="2020-07-27T05:51:05Z" level=warning msg="skip unknown headers: unknown"
> time="2020-07-27T05:51:05Z" level=warning msg="skip unknown headers: unknown"


Workaround:
Executing following command on the user profile created in AWS cognito this will set the email verified attribute to true and thus authzadaptor would return fine for the user.

`aws cognito-idp admin-update-user-attributes --user-pool-id *** --username *** --user-attributes '[{"Name": "email_verified", "Value": "true"}]'`

As part of this, I request to update documentation to include a Note for this scenario in the AWS authentication section.
